docs/compilers/CSharp/Runtime Async Design.md

```diff
@@ -27,20 +27,32 @@ We use the following helper APIs to indicate suspension points to the runtime, i
 ```cs
 namespace System.Runtime.CompilerServices;
 
-// These methods are used to await things that cannot use runtime async signature form
 // TODO: Clarify which of these should be preferred? Should we always emit the `Unsafe` version when awaiting something that implements `ICriticalNotifyCompletion`?
 namespace System.Runtime.CompilerServices;
 
 public static class RuntimeHelpers
 {
-    [RuntimeAsyncMethod]
+    // These methods are used to await things that cannot use the Await helpers below
+    [MethodImpl(MethodImplOptions.Async)]
     public static Task AwaitAwaiterFromRuntimeAsync<TAwaiter>(TAwaiter awaiter) where TAwaiter : INotifyCompletion;
-    [RuntimeAsyncMethod]
+    [MethodImpl(MethodImplOptions.Async)]
     public static Task UnsafeAwaitAwaiterFromRuntimeAsync<TAwaiter>(TAwaiter awaiter) where TAwaiter : ICriticalNotifyCompletion;
+
+    // These methods are used to directly await method calls
+    // TODO: Do we only use these when directly awaiting a method call, or do we always use these when a `Task` is awaited, even if that task is a local?
```

Comment:
It's up to you. `Await` will result in smaller IL over expanding the awaiter based pattern and using `AwaitAwaiterFromRuntimeAsync`.

Reply:
It's mainly a question for y'all :laughing:. If you're ok with the pattern being directly calling `Await`, I'm good with it as well.

Reply:
I don't think there will be a difference perf wise (well, if there is it should be on the JIT to optimize it), so I think going with the smaller IL expansion is preferable.